src/App.tsx

```diff
@@ -1,223 +1,171 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useMemo,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterType } from './types/FilterValues';
+import { NewTodoField } from './components/NewTodoField/NewTodoField';
+import { TodoItem } from './components/TodoItem/TodoItem';
+import { ErrorMessage } from './types/ErrorMessage';
+
+const defaultTodo = {
+  id: 0,
+  userId: 0,
+  title: '',
+  completed: false,
+};
 
-export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const App = (): JSX.Element | null => {
   const user = useContext(AuthContext);
+
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage]
+    = useState<ErrorMessage>(ErrorMessage.None);
+  const [filter, setFilter] = useState(FilterType.All);
+  const [isAdding, setIsAdding] = useState(false);
+  const [temporaryTodo, setTemporaryTodo] = useState(defaultTodo);
+  const [completedIsRemoving, setCompletedIsRemoving] = useState(false);
+
+  const numberOfTodos = useMemo(() => {
+    return todos.filter((todo) => !todo.completed).length;
+  }, [todos]);
+
+  const numberOfCompleted = useMemo<number>(() => {
+    return [...todos].filter((todo) => todo.completed).length;
+  }, [todos]);
+
+  const visibleTodos = useMemo<Todo[]>(() => {
+    if (todos) {
+      return [...todos].filter(todo => {
+        switch (filter) {
+          case FilterType.Active:
+            return !todo.completed;
+
+          case FilterType.Completed:
+            return todo.completed;
+
+          case FilterType.All:
+          default:
+            return true;
+        }
+      });
+    }
+
+    return [];
+  }, [todos, filter]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    setErrorMessage(ErrorMessage.None);
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
-  }, []);
+  }, [user]);
+
+  useEffect(() => {
+    if (user) {
+      getTodos(user.id)
+        .then(result => setTodos(result))
+        .catch(() => {
+          setErrorMessage(ErrorMessage.CannotLoadTodos);
+        });
+    }
+  }, [todos]);
```

Comment:
You have todos as dependecy of useEffect which set new todos. It results in recursion call.
![image](https://user-images.githubusercontent.com/102293862/200632458-0018fb26-52ac-4f7b-a80c-61c939e840e0.png)